pkg/service/helper.go

```diff
@@ -22,27 +27,74 @@ func diskAttachmentByVmAndDisk(connection *ovirtsdk.Connection, vmId string, dis
 			return attachment, nil
 		}
 	}
-	return nil, &AttachmentNotFoundError{
-		vmId:   vmId,
-		diskId: diskId,
-	}
+	return nil, nil
 }
 
-func diskAttachmentByDisk(connection *ovirtsdk.Connection, diskId string) (*ovirtsdk.DiskAttachment, *ovirtsdk.DiskAttachmentService, error) {
-	vmsService := connection.SystemService().VmsService()
-	for _, vm := range vmsService.List().MustSend().MustVms().Slice() {
-		vmService := connection.SystemService().VmsService().VmService(vm.MustId())
-		attachments, err := vmService.DiskAttachmentsService().List().Send()
+func findDiskAttachmentByDiskInVMs(conn *ovirtsdk.Connection, diskId string, vms []*ovirtsdk.Vm) (*ovirtsdk.DiskAttachment, error) {
+	for _, vm := range vms {
+		diskAttachment, err := diskAttachmentByVmAndDisk(conn, vm.MustId(), diskId)
 		if err != nil {
-			return nil, nil, err
+			klog.Error(err)
+			return nil, errors.Wrap(err, "failed finding disk attachments")
+		}
+		if diskAttachment != nil {
+			return diskAttachment, nil
 		}
+	}
+	return nil, nil
+}
 
-		for _, attachment := range attachments.MustAttachments().Slice() {
-			if attachment.MustId() == diskId {
-				return attachment, vmService.DiskAttachmentsService().AttachmentService(attachment.MustId()), nil
+func findDiskAttachmentByDiskInCluster(ctx context.Context, c client.Client, conn *ovirtsdk.Connection, diskId string) (*ovirtsdk.DiskAttachment, error) {
+	tag, err := getClusterTag(ctx, c)
+	if err != nil {
+		klog.Error(err)
+		return nil, errors.Wrap(err, "failed finding openshift cluster tag")
+	}
+	vms, err := getVmsWithTag(conn, tag)
+	if err != nil {
+		klog.Error(err)
+		return nil, errors.Wrap(err, "failed searching VMs")
+	}
+	return findDiskAttachmentByDiskInVMs(conn, diskId, vms)
+}
 
-			}
-		}
+func getVmsWithTag(connection *ovirtsdk.Connection, tag string) ([]*ovirtsdk.Vm, error) {
+	searchTerm := fmt.Sprintf("tag=%s", tag)
+	vmsWithTag, err := connection.SystemService().VmsService().List().Search(searchTerm).Send()
+	if err != nil {
+		return nil, fmt.Errorf("faild listing VMs with tag %s, error: %w", tag, err)
+	}
+	return vmsWithTag.MustVms().Slice(), nil
+}
+
+func getClusterTag(ctx context.Context, c client.Client) (string, error) {
+	infra := &configv1.Infrastructure{}
+	objKey := client.ObjectKey{Name: globalInfrastuctureName}
+	if err := c.Get(ctx, objKey, infra); err != nil {
+		return "", errors.Wrap(err, "error getting infrastucture data")
+	}
+	return infra.Status.InfrastructureName, nil
+}
+
+func expandDiskByDiskDiskAttachment(conn *ovirtsdk.Connection, diskAttachment *ovirtsdk.DiskAttachment, bytes int64) error {
+	diskAttachment.MustDisk().SetProvisionedSize(bytes)
+	_, err := conn.SystemService().VmsService().VmService(diskAttachment.MustVm().MustId()).
+		DiskAttachmentsService().AttachmentService(diskAttachment.MustId()).
+		Update().DiskAttachment(diskAttachment).Send()
+	if err != nil {
```

Comment:
I think this part is not trivial, because disk update is an async operation in ovirt-engine, so we may need to introduce a waiting mechanism here that will check the status of the job (finished/failed) and then continue

we have this in OST[1], a simple loop that checks the result of get_jobs_statuses[2], but we need to pass a correlation_id

I also did something similar in the terraform provider[3]

[1] https://github.com/oVirt/ovirt-system-tests/blob/master/basic-suite-master/test-scenarios/test_004_basic_sanity.py#L1558
[2] https://github.com/oVirt/ovirt-system-tests/blob/master/ost_utils/ost_utils/engine_utils.py#L48
[3] https://github.com/oVirt/terraform-provider-ovirt/pull/242/files